### ISSUE TITLE
inline configurations (mypy comments) in strings still work

### ISSUE THREAD

**Issue by wyattscarpenter** [python/mypy#19441]
**Bug Report**

I take it the [mypy inline configuration comments](https://mypy.readthedocs.io/en/stable/inline_config.html) are supposed to work like regular hot comments, and not just activate whenever they appear as characters in the source file, such as in a string. However, as I discovered while investigating https://github.com/python/mypy/pull/17875#issuecomment-2939335586, they aren't actually parsed according to the Python syntax rules (because this would be slower and more complicated, at least given our current implementation) and therefore they do work if they're in a string.

For example, in the following code, untyped defs are disallowed and an error is raised:

```python
"""
This is just a silly little multi-line string literal, surely it couldn't—
# mypy: disallow-untyped-defs
"""
def foo():
    return 1
```

**Your Environment**

- Mypy version used: this was probably 1.15, but I'd be surprised if it's been fixed by 1.17 (current as of time of writing)

**Related**

This, #12358, and https://github.com/python/mypy/issues/19366 form a loose family of "things we could easily fix if we had a parser that told us about comments to begin with" ie if https://github.com/python/cpython/issues/101494#issuecomment-2395119743 were implemented.
